### PR TITLE
Documented the process for Azure Cloud Provider for Rancher-provisioned RKE2 clusters

### LINKED ISSUE #3505: [Rancher2] Document the process for Azure Cloud Provider for rancher-provisioned RKE2 clusters
**Request Summary:**
As of v2.6.0 release and the advent of rke2 (preview) support, we should document the process for configuring the azure cloud provider. It's very different from RKE1 and not trivial.

**Details:**
I believe this process would entail updating https://rancher.com/docs/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/ such that there is a section for RKE1 and one for RKE2 then add instructions like below for RKE2. However this document doesn't appear accurate as well - it needs some refactoring.

For RKE2 the process is like follows (please do not use my screenshots as they are not perfect). I have tried to provide a step by step process that could be wordsmithed as needed.

1. Choose "Azure" from the Cloud Provider drop down in the Cluster Configuration section.
image
2. Supply the Cloud Provider Config. An example is provided below. Please modify it as necessary to fit your needs. Please note that Rancher will automatically create a new Network Security Group, Resource Group, Availability Set, Subnet, and Virtual Network. If you already have some or all of these created, you will need to specify them before creating the cluster. You can click on "Show Advanced" to see more of these automatically generated names and update them if necessary. Your Cloud Provider Config MUST match the fields plugged in in the Machine Pools section. If you have multiple pools, they must all use the same Resource Group, Availability Set, Subnet, Virtual Network, and Network Security Group.
```
{
    "cloud":"AzurePublicCloud",
    "tenantId": "YOUR TENANTID HERE",
    "aadClientId": "YOUR AADCLIENTID HERE",
    "aadClientSecret": "YOUR AADCLIENTSECRET HERE",
    "subscriptionId": "YOUR SUBSCRIPTIONID HERE",
    "resourceGroup": "docker-machine",
    "location": "westus",
    "subnetName": "docker-machine",
    "securityGroupName": "rancher-managed-KA4jV9V2",
    "securityGroupResourceGroup": "docker-machine",
    "vnetName": "docker-machine-vnet",
    "vnetResourceGroup": "docker-machine",
    "primaryAvailabilitySetName": "docker-machine",
    "routeTableResourceGroup": "docker-machine",
    "cloudProviderBackoff": false,
    "useManagedIdentityExtension": false,
    "useInstanceMetadata": true
}
```
3. Under the Cluster Configuration > Advanced section, click "Add" under Additional Controller Manager Args and add this flag:
`--configure-cloud-routes=false`
image
image
4. Click the Create button to submit the form and create the cluster.

### PR DESCRIPTION
**Pull request by jtravee**
Closes #3505

### CODE REVIEW

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 1]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
 Only hosts expected to be load balancer back ends need to be in this group.
+
+# RKE2 Cluster Set-up in Rancher
+
+1. Choose "Azure" from the Cloud Provider drop-down in the Cluster Configuration section.
+
+1. * Supply the Cloud Provider Configuration. Note that Rancher will automatically create a new Network Security Group, Resource Group, Availability Set, Subnet, and Virtual Network. If you already have some or all of these created, you will need to specify them before creating the cluster. 
+   * You can click on "Show Advanced" to see more of these automatically generated names and update them if
+   necessary. Your Cloud Provider Configuration **must** match the fields in the Machine Pools section. If you have multiple pools, they must all use the same Resource Group, Availability Set, Subnet, Virtual Network, and Network Security Group. 
+   * If you choose to edit as YAML, an example is provided below. You will modify it as needed.
```
Comment (btat) [inline content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:81]:
```suggestion
   * An example is provided below. You will modify it as needed.
```

Looks like the example provided is JSON

**Review by btat — CHANGES_REQUESTED** [review]

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 2]
```diff
@@ -16,7 +16,9 @@ The following account types are not supported for Azure Subscriptions:
 - Single tenant accounts (i.e. accounts with no subscriptions).
 - Multi-subscription accounts.
 
-To set up the Azure cloud provider following credentials need to be configured:
+# Prerequisites for RKE and RKE2
+
+**To set up the Azure cloud provider in both RKE and RKE2, the following credentials need to be configured:**
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:21]: Revised as suggested.

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 3]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
+    "routeTableResourceGroup": "docker-machine",
+    "cloudProviderBackoff": false,
+    "useManagedIdentityExtension": false,
+    "useInstanceMetadata": true
+    ```
+    {{% /accordion %}}
+    
+1. Under the Cluster Configuration > Advanced section, click "Add" under Additional Controller Manager Args and add this flag: `--configure-cloud-routes=false`
+
+1. Click the Create button to submit the form and create the cluster.
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:111]: Revised as suggested.

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 4]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
+    "vnetResourceGroup": "docker-machine",
+    "primaryAvailabilitySetName": "docker-machine",
+    "routeTableResourceGroup": "docker-machine",
+    "cloudProviderBackoff": false,
+    "useManagedIdentityExtension": false,
+    "useInstanceMetadata": true
+    ```
+    {{% /accordion %}}
+    
+1. Under the Cluster Configuration > Advanced section, click "Add" under Additional Controller Manager Args and add this flag: `--configure-cloud-routes=false`
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:109]: Updated as suggested.

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 5]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
+1. * Supply the Cloud Provider Configuration. Note that Rancher will automatically create a new Network Security Group, Resource Group, Availability Set, Subnet, and Virtual Network. If you already have some or all of these created, you will need to specify them before creating the cluster. 
+   * You can click on "Show Advanced" to see more of these automatically generated names and update them if
+   necessary. Your Cloud Provider Configuration **must** match the fields in the Machine Pools section. If you have multiple pools, they must all use the same Resource Group, Availability Set, Subnet, Virtual Network, and Network Security Group. 
+   * If you choose to edit as YAML, an example is provided below. You will modify it as needed.
+
+    {{% accordion id="v2.6.0-cloud-provider-config-file" label="Example Cloud Provider Config File" %}}
+
+    ```yaml
+
+    # Cloud Provider Config
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:87]: Done.

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 6]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
+# RKE2 Cluster Set-up in Rancher
+
+1. Choose "Azure" from the Cloud Provider drop-down in the Cluster Configuration section.
+
+1. * Supply the Cloud Provider Configuration. Note that Rancher will automatically create a new Network Security Group, Resource Group, Availability Set, Subnet, and Virtual Network. If you already have some or all of these created, you will need to specify them before creating the cluster. 
+   * You can click on "Show Advanced" to see more of these automatically generated names and update them if
+   necessary. Your Cloud Provider Configuration **must** match the fields in the Machine Pools section. If you have multiple pools, they must all use the same Resource Group, Availability Set, Subnet, Virtual Network, and Network Security Group. 
+   * If you choose to edit as YAML, an example is provided below. You will modify it as needed.
+
+    {{% accordion id="v2.6.0-cloud-provider-config-file" label="Example Cloud Provider Config File" %}}
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:83]: Done.

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 7]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
+
+    ```yaml
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:85]: Done.

File: content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md [hunk 8]
```diff
@@ -68,3 +70,42 @@ If you provision hosts using Rancher Machine Azure driver, you will need to edit
...
+    "securityGroupName": "rancher-managed-KA4jV9V2",
+    "securityGroupResourceGroup": "docker-machine",
+    "vnetName": "docker-machine-vnet",
+    "vnetResourceGroup": "docker-machine",
+    "primaryAvailabilitySetName": "docker-machine",
+    "routeTableResourceGroup": "docker-machine",
+    "cloudProviderBackoff": false,
+    "useManagedIdentityExtension": false,
+    "useInstanceMetadata": true
+    ```
```
[the remark this answers is not in the document]
Comment (jtravee) [reply content/rancher/v2.6/en/cluster-provisioning/rke-clusters/cloud-providers/azure/_index.md:106]: Done. Initially, the indented backticks were needed to ensure the numbering below the code block was correct. But in this case, I believe inserting the { } enables the numbering to proceed correctly without needing indention.

**Pull request merged**